src/utils/components/DiskModal/components/StorageProfileSettings/VolumeMode.tsx

```diff
@@ -1,58 +1,96 @@
-import React, { FC, useEffect, useMemo } from 'react';
-import { Controller, useFormContext } from 'react-hook-form';
+import React, { FC, useCallback, useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { Trans } from 'react-i18next';
+import { Link } from 'react-router-dom-v5-compat';
+import { uniq } from 'lodash';
 
-import {
-  V1beta1StorageSpecAccessModesEnum,
-  V1beta1StorageSpecVolumeModeEnum,
-} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { V1beta1StorageSpecVolumeModeEnum } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import HelpTextIcon from '@kubevirt-utils/components/HelpTextIcon/HelpTextIcon';
+import { documentationURL } from '@kubevirt-utils/constants/documentation';
 import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { ClaimPropertySets } from '@kubevirt-utils/types/storage';
 import { FormGroup, Radio } from '@patternfly/react-core';
 
 import { V1DiskFormState } from '../../utils/types';
-import {
-  ACCESS_MODE_FIELD,
-  STORAGE_CLASS_PROVIDER_FIELD,
-  VOLUME_MODE_FIELD,
-  VOLUMEMODE_FIELDID,
-} from '../utils/constants';
-import { getVolumeModeForProvisioner, VOLUME_MODE_RADIO_OPTIONS } from '../utils/modesMapping';
+import { ACCESS_MODE_FIELD, VOLUME_MODE_FIELD, VOLUMEMODE_FIELDID } from '../utils/constants';
+import { getAccessModesForVolume, VOLUME_MODE_RADIO_OPTIONS } from '../utils/modesMapping';
 
-const VolumeMode: FC = () => {
+import RecommendationLabel from './RecommendationLabel';
+
+import './ApplyStorageProfileSettings.scss';
+type VolumeModeProps = {
+  claimPropertySets: ClaimPropertySets;
+};
+
+const VolumeMode: FC<VolumeModeProps> = ({ claimPropertySets }) => {
   const { t } = useKubevirtTranslation();
 
-  const { control, setValue, watch } = useFormContext<V1DiskFormState>();
+  const { setValue, watch } = useFormContext<V1DiskFormState>();
 
   const volumeMode = watch(VOLUME_MODE_FIELD);
-  const accessModes = watch(ACCESS_MODE_FIELD) as V1beta1StorageSpecAccessModesEnum[];
-  const storageClassProvisioner = watch(STORAGE_CLASS_PROVIDER_FIELD);
+  const recommendedVolumeModes = uniq(
```

Comment:
why uniq? u can use set, its string values

Reply:
We've already discussed this in https://github.com/kubevirt-ui/kubevirt-plugin/pull/2524#discussion_r2007942877
Basically we need to keep the initial order.